server/src/main/java/org/elasticsearch/common/file/AbstractFileWatchingService.java

```diff
@@ -363,8 +365,14 @@ final WatchKey enableDirectoryWatcher(WatchKey previousKey, Path settingsDir) th
         } while (true);
     }
 
-    // package private for testing
-    void processSettingsAndNotifyListeners() throws InterruptedException {
+    /**
+     * Process settings and notify listeners
+     *
+     * <p>Logs an error if the file can't be read or if changes can't be applied
+     *
+     * @throws InterruptedException if the file processing is interrupted by another thread.
+     */
+    public void processSettingsAndNotifyListeners() throws InterruptedException {
```

Comment:
I do not think this should be public. Why can't it remain package private since this PR adds test in this same new package?

Reply:
It was mocked in tests for `FileSettingsService` but I found another way the tests can work.